e2e-tests/test/zks-apis.test.ts

```diff
@@ -74,3 +74,25 @@ describe("zks_getTransactionDetails", function () {
     expect(details["initiatorAddress"].toLowerCase()).to.equal(wallet.address.toLowerCase());
   });
 });
+
+describe("zks_getBytecodeByHash", function () {
+  it("Should stored bytecode at address", async function () {
+    // Arrange
+    const wallet = new Wallet(RichAccounts[0].PrivateKey);
+    const deployer = new Deployer(hre, wallet);
+    const artifact = await deployer.loadArtifact("Greeter");
+    const greeter = await deployContract(deployer, "Greeter", ["Hi"]);
+    expect(await greeter.greet()).to.eq("Hi");
+    // get the bytecode hash from the KnownCodesStorage log
+    const logs = await provider.send("eth_getLogs", [{ address: "0x0000000000000000000000000000000000008004" }]);
+    expect(logs).to.not.be.empty;
+    expect(logs[logs.length - 1].topics).to.have.lengthOf(3);
+    const bytecodeHash = logs[0].topics[1];
```

Comment:
I believe the topics are not in order... so unsure how to validate consistently. Especially when previous transactions have occurred 😞

Reply:
I adapted the test to only use the contract's deployed block to fetch the logs from. The topics are always in order. The first topic is the solidity's event's keccak hash and the subsequent ones are the parameters.